### PR TITLE
Refactor to use guzzle

### LINKED ISSUE #146: Refactor to use guzzle

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Closes #146.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
I'll update the upgrading PR with the changes shortly.

File: tests/ConfigurationTest.php [hunk 1]
```diff
@@ -23,17 +23,6 @@ public function testDoesNotAcceptBadApiKey()
         new Configuration([]);
     }
 
-    public function testDefaultEndpoint()
-    {
-        $this->assertSame($this->config->getNotifyEndpoint(), 'https://notify.bugsnag.com');
```
Comment (kattrali) [inline tests/ConfigurationTest.php]: There should probably be an equivalent test to ensure the default guzzle configuration points to the endpoint.

Comment (GrahamCampbell) [inline tests/ConfigurationTest.php]: No need. Guzzle already has tests for this. We know it works.

Comment (GrahamCampbell) [inline tests/ConfigurationTest.php]: https://github.com/guzzle/guzzle/blob/master/tests/ClientTest.php#L57-L72

Comment (GrahamCampbell) [inline tests/ConfigurationTest.php]: User can set the base uri by calling `$client->getGuzzle()->something else i can't remember`.

Comment (kattrali) [inline tests/ConfigurationTest.php]: oh, I mean testing that our default instance base uri == `https://notify.bugsnag.com`. A typo in our constant or something should be caught.

Comment (GrahamCampbell) [inline tests/ConfigurationTest.php]: Ok, sure. I can add that.

Comment (GrahamCampbell) [inline tests/ConfigurationTest.php]: Done.

**Comment by kattrali** [comment]
🎉  Awesome work!

**Pull request merged**